src/main/java/duke/time/TimePoint.java

```diff
@@ -0,0 +1,64 @@
+package duke.time;
+
+import duke.exception.DukeInputException;
+
+/**
+ * Representation for a point in time in text.
+ */
+public class TimePoint {
+
+    /**
+     * String information of time represented by this <code>TimePoint</code>.
+     */
+    private String timePoint;
+
+    /**
+     * Creates an empty <code>TimePoint</code>.
+     */
+    public TimePoint() {
+        this.timePoint = "";
+    }
+
+    private TimePoint(String timePoint) {
+        this.timePoint = timePoint;
+    }
+
+    /**
+     * Factory method for creating <code>TimePoint</code>s from a given string input.
+     * Attempts to create a <code>DateTime</code> representation of the time description provided.
+     * If unable to parse provided description, a String-based representation is created instead.
+     *
+     * @param timeInfoString Time representation input to be parsed.
+     * @return <code>TimePoint</code> object representing inputted time.
+     */
+    public static TimePoint of(String timeInfoString) {
```

Comment:
the name "of" is slightly ambiguous, might I suggest something like parseTime?

Reply:
Renamed to `parseTime` as suggested.